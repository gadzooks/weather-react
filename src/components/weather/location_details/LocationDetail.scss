.location-details-page {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  max-width: 600px;

  .heading {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    width: 600px;
  }
}

div.location-details-div {
  padding-top: 10px;
}

table.location-details {
  border-collapse:collapse;
  font-size: 1.2rem;

  tbody {
    white-space: nowrap;
  }

  thead tr {
    background-color: #cfe2f5;
    text-align: center;
  }

  tbody td {
    padding: 6px 8px;
  }

  tr.heading th {
    font-weight: bold;
    text-align: center;
    background-color: #f2f7fb;
  }

  tr.secondary-heading th {
    background-color: #ffeaaa;
  }

  td.align-right {
    text-align: right;
  }

  .center {
    text-align: center;
  }
}

.border-right {
  border-right: 1px solid #afafaf;
}

@media (max-width: 500px) {
//   table.location-details tbody tr td {
//     padding: 6px 9px;
//   }
.location-details-page {
//   max-width: 400px;
}
  
}