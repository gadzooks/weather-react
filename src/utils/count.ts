import { AlertsById } from '../interfaces/ForecastResponseInterface';

export default function alertsFound(alertsById: AlertsById | null): boolean {
  if (alertsById === null) {
    return false;
  }
  return JSON.stringify(alertsById) !== '{}';
}
