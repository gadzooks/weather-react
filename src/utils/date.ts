import { format } from 'fecha';

export function formatDate(str: string) {
  return new Date(str)
    .toLocaleDateString([], { year: 'numeric', month: 'long', day: 'numeric' });
}

export function calculateWeekends(dates: (Date | null)[]): boolean[] {
  const weekends: boolean[] = [];
  if (!dates) {
    return weekends;
  }

  dates.forEach((d) => {
    if (d) {
      const dayOfWeek = format(d, 'ddd').toUpperCase();
      weekends.push(dayOfWeek === 'SUN' || dayOfWeek === 'SAT');
    } else {
      weekends.push(false);
    }
  });

  return weekends;
}

export function dateDifferenceInDays(epochSeconds: number) {
  if (epochSeconds === undefined) {
    return null;
  }

  const date = new Date(epochSeconds * 1_000);
  const today = new Date();

  const diff = (date.getTime() - today.getTime()) / (1000 * 3600 * 24);
  return Math.floor(diff + 1);
}
