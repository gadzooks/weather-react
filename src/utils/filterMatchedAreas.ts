import { RegionsById } from '../interfaces/ForecastResponseInterface';
import { MatchedAreas } from '../interfaces/MatchedAreas';

export default function findMatchedAreas(
  needle: RegExp | null,
  regionsById: RegionsById,
): MatchedAreas {
  // console.log(`matchedLocations called with ${needle}`);
  const matchedAreas: MatchedAreas = {
    totalMatchedRegions: 0,
  };

  regionsById.allIds.forEach((regionName) => {
    const region = regionsById.byId[regionName];
    if (needle) {
      const locations = region.locations.filter((l) => l.description.match(needle));
      if (locations.length > 0) {
        matchedAreas.regions ||= [];
        matchedAreas.locationsByRegion ||= {};
        matchedAreas.regions.push(region);
        matchedAreas.locationsByRegion[region.name] = locations;
      }
    } else {
      matchedAreas.regions ||= [];
      matchedAreas.regions.push(region);
      matchedAreas.locationsByRegion ||= {};
      matchedAreas.locationsByRegion[region.name] = region.locations;
    }
  });
  matchedAreas.totalMatchedRegions = matchedAreas.regions?.length || 0;
  return matchedAreas;
}
