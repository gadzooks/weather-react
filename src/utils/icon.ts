const iconMapping: {[key: string] : string} = {
  'clear-day': 'day-sunny',
  'clear-night': 'night-clear',
  rain: 'rain',
  snow: 'snow',
  sleet: 'sleet',
  wind: 'strong-wind',
  fog: 'fog',
  cloudy: 'cloudy',
  'partly-cloudy-day': 'cloudy',
  'partly-cloudy-night': 'night-partly-cloudy',
  hail: 'hail',
  thunderstorm: 'thunderstorm',
  tornado: 'tornado',
};

// eslint-disable-next-line max-len
export default function iconClass(icon: string, precip: number, cloudCover: number, maxTemp: number): string {
  let mapping = iconMapping[icon] || 'na';
  let additionalClass = '';

  if (mapping === 'cloudy') {
    if (precip) {
      if (precip < 30) {
        additionalClass = 'sunshine-10';
      } else if (precip < 60) {
        additionalClass = 'sunshine-50';
      } else {
        additionalClass = 'sunshine-100';
      }
    }

    // OVERRIDE sunshine color to greyish if cloud cover is high
    if (cloudCover > 60) additionalClass = 'sunshine-50';

    // show a bit of sun peeking out if there is less than 50% cloud cover
    if (cloudCover <= 50) mapping = 'day-cloudy';
  } else if (mapping === 'day-sunny' && maxTemp >= 80) {
    mapping = 'hot';
  }

  if (maxTemp >= 90) {
    additionalClass += ' high-temp-wi-hotter';
  } else if (maxTemp >= 80) {
    additionalClass += ' high-temp-wi-hot';
  }

  return `wi weather-icon wi-${mapping} ${mapping} ${additionalClass}`;
}

export function precipitation(precip: number): string {
  if (!precip) return '';

  let computedPrecip = precip * 100;
  if (computedPrecip > 100) computedPrecip /= 100;
  return `${Math.round(computedPrecip)}%`;
}

export function precipitationAmount(p: number): string {
  if (p) {
    if (p === 0 || Math.round(p) === 0) return '-';
    return `${Math.round(p)}"`;
  }
  return '-';
}
