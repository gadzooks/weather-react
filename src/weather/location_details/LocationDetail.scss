table.location-details {
    tbody {
        white-space: nowrap;
    }

    width: 50%;

    tr.heading th {
        font-weight: bold;
        text-align: center;
        background-color: #f2f7fb;
    }

    tr.secondary-heading th {
        background-color: #ffeaaa;
    }

    td.align-right {
        text-align: right;
    }
}

.center {
    text-align: center;
}

.border-right {
    border-right: 1px solid #afafaf;
}

