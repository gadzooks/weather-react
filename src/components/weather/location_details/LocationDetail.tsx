import React from 'react';
import {
  Link,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
} from '@mui/material';
import { format } from 'fecha';
import '../../../css/weather-icons.css';
import './LocationDetail.scss';
import convertToSentence from '../../../utils/string';
import { DailyForecastInterface } from '../../../interfaces/DailyForecastInterface';
import { LocationInterface } from '../../../interfaces/LocationInterface';
import { RegionInterface } from '../../../interfaces/RegionInterface';
import WtaLink from './WtaLink';
import WeatherIcon from '../main_page/WeatherIcon';

interface LocationDetailProps {
  region: RegionInterface;
  location: LocationInterface;
  forecast: DailyForecastInterface[];
  dates: (Date | null)[];
  isWeekend: boolean[];
}

function LocationDetail(props: LocationDetailProps) {
  const { region } = props;
  const { location } = props;
  const { description } = location;
  const { forecast } = props;
  const { dates } = props;
  const { isWeekend } = props;
  return (
    <div id={location.name}>
      <Table className='location-details'>
        <TableHead>
          <TableRow className='heading'>
            <TableCell colSpan={8} className='heading'>
              {`${description.toUpperCase()}  `}
              <WtaLink
                wtaRegion={region.searchKey}
                wtaSubRegion={location.sub_region}
              />
              <Link href='#top'>(top)</Link>
            </TableCell>
          </TableRow>
          <TableRow className='secondary-heading'>
            <TableCell colSpan={2} className='center border-right'>
              DATE
            </TableCell>
            <TableCell colSpan={2} className='center border-right'>
              DETAILS
            </TableCell>
            <TableCell className='border-right'>LOW / HIGH</TableCell>
            <TableCell colSpan={2} className='center border-right'>
              PRECIP
            </TableCell>
            <TableCell>CLOUD COV</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {forecast.map((row, id) => {
            const d = dates[id];
            const weekendClassName = isWeekend[id] ? ' weekend ' : ' ';
            // console.log([row, id]);
            if (d) {
              return (
                <TableRow className={weekendClassName} key={row.datetime}>
                  <TableCell>{format(d, 'ddd').toUpperCase()}</TableCell>
                  <TableCell className='border-right'>
                    {format(d, 'MMM Do').toUpperCase()}
                  </TableCell>
                  <TableCell>
                    <WeatherIcon {...row} key={row.datetime} />
                  </TableCell>
                  <TableCell className='border-right'>
                    {convertToSentence(row.icon)}
                  </TableCell>
                  <TableCell className='border-right'>
                    {`${Math.round(
                      row.tempmin,
                    )}° / ${Math.round(row.tempmax)}°`}

                  </TableCell>
                  <TableCell className='align-right'>
                    {`${Math.round(
                      row.precipprob,
                    )}%`}

                  </TableCell>
                  <TableCell className='align-right border-right'>
                    {`${Math.round(
                      row.precip,
                    )}"`}

                  </TableCell>
                  <TableCell className='align-right'>{`${row.cloudcover}%       `}</TableCell>
                </TableRow>
              );
            }
            return null;
          })}
        </TableBody>
      </Table>
    </div>
  );
}

export default LocationDetail;

// forecasts: {
//     byId: {
//       renton: [
//         {
//           time: "2021-04-17T07:00:00.000+00:00",
//           summary: "Clear conditions throughout the day.",
//           icon: "day-hail",
//           precipProbability: 0.0,
//           temperature: 60.9,
//           apparentTemperature: 60.9,
//           dewPoint: 40.1,
//           visibility: 120,
//           cloudCover: 1,
//           temperatureHigh: 74.9,
//           temperatureLow: 47.3,
//           sunsetTime: 1618714951,
//           sunriseTime: 1618665359,
//           precipAmount: 0.0,
//         },
