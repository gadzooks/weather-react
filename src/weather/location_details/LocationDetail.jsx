import { Table, TableBody, TableCell, TableHead, TableRow } from "@mui/material";
import { format } from 'fecha';
import WeatherIcon from "../WeatherIcon";
import '../../css/weather-icons.css'
import './LocationDetail.scss'
import { convertToSentence } from "../../utils/string";

function LocationDetail(props) {
    const name = props.name;
    const forecast = props.forecast;
    const dates = props.dates;
    const isWeekend = props.isWeekend;
    return (
        <>
            <Table className='location-details'>
                <TableHead>
                    <TableRow className="heading">
                        <TableCell colSpan={8} className="heading">
                            {name.toUpperCase()}
                        </TableCell>
                    </TableRow>
                    <TableRow className="secondary-heading">
                        <TableCell colSpan={2} className="center border-right">DATE</TableCell>
                        <TableCell colSpan={2} className="center border-right">DETAILS</TableCell>
                        <TableCell className="border-right">HI / LOW</TableCell>
                        <TableCell colSpan={2} className="center border-right">PRECIP</TableCell>
                        <TableCell>CLOUD COV</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {forecast.map((row, id) => {
                        const d = dates[id];
                        const weekendClassName = isWeekend[id] ? ' weekend ' : ' ';
                        // console.log([row, id]);
                        return <TableRow className={weekendClassName} key={id}>
                            <TableCell >{format(d, 'ddd').toUpperCase()}</TableCell>
                            <TableCell className="border-right">{format(d, 'MMM Do').toUpperCase()}</TableCell>
                            <TableCell>
                                <WeatherIcon d={row} index={id} />
                            </TableCell>
                            <TableCell className="border-right">{convertToSentence(row.icon)}</TableCell>
                            <TableCell className="border-right">{Math.round(row.tempmax) + '° / ' + Math.round(row.tempmin) + '°'}</TableCell>
                            <TableCell className="align-right">{Math.round(row.precipprob) + '%'}</TableCell>
                            <TableCell className="align-right border-right">{Math.round(row.precip) + '"'}</TableCell>
                            <TableCell className="align-right">{row.cloudcover + '%       '}</TableCell>
                        </TableRow>
                    })}
                </TableBody>
            </Table>
        </>
    );
}

export default LocationDetail;

// forecasts: {
//     byId: {
//       renton: [
//         {
//           time: "2021-04-17T07:00:00.000+00:00",
//           summary: "Clear conditions throughout the day.",
//           icon: "day-hail",
//           precipProbability: 0.0,
//           temperature: 60.9,
//           apparentTemperature: 60.9,
//           dewPoint: 40.1,
//           visibility: 120,
//           cloudCover: 1,
//           temperatureHigh: 74.9,
//           temperatureLow: 47.3,
//           sunsetTime: 1618714951,
//           sunriseTime: 1618665359,
//           precipAmount: 0.0,
//         },