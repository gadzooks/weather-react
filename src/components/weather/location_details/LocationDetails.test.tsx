import { parse } from 'fecha';
import React from 'react';
import renderer from 'react-test-renderer';
import mockWeatherForecastNormalized from '../../../api/mockData';
import {
  ForecastDates,
  ForecastResponse,
  ForecastResponseStatus,
} from '../../../interfaces/ForecastResponseInterface';
import { calculateWeekends } from '../../../utils/date';
import LocationDetail, { LocationDetailProps } from './LocationDetail';

const { ResizeObserver } = window;

// from https://github.com/maslianok/react-resize-detector/issues/145#issuecomment-953569721
beforeEach(() => {
  // @ts-ignore
  delete window.ResizeObserver;
  window.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
  }));
});

afterEach(() => {
  window.ResizeObserver = ResizeObserver;
  jest.restoreAllMocks();
});

it('renders a snapshot', () => {
  const forecast = mockWeatherForecastNormalized()
    .data as unknown as ForecastResponse;
  const parsedDates = forecast.dates.map((d: string) => parse(d, 'YYYY-MM-DD'));
  const weekends = calculateWeekends(parsedDates);

  const forecastDates: ForecastDates = {
    dates: forecast.dates,
    parsedDates,
    weekends,
  };

  const appState: ForecastResponseStatus = {
    isLoaded: true,
    error: null,
    forecast,
  };

  const locDetailProps: LocationDetailProps = {
    appState,
    forecastDetailsForLocation:
      '{"name":"seattle","description":"Cle Elum Area1","latitude":47.196943,"longitude":-120.939134,"sub_region":"767d751df8b6495999e96486d4d32d49","region":"snowqualmie_region","wtaRegionKey":"04d37e830680c65b61df474e7e655d64"}',
    setForecastDetailsForLocation: () => {},
    forecastDates,
  };

  const tree = renderer
    .create(<LocationDetail {...locDetailProps} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
