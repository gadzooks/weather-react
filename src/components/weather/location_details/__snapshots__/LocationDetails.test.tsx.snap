// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders a snapshot 1`] = `
<div
  className="location-details-page"
>
  <div
    className="heading"
  >
    <div>
      <a
        className=""
        href="https://www.wta.org/go-outside/trip-reports/tripreport_search?title=&searchabletext=&author=&startdate=&_submit=&enddate=&_submit=&month=all&format=list&filter=Search&region=04d37e830680c65b61df474e7e655d64&subregion=767d751df8b6495999e96486d4d32d49"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="wta-logo"
          className="wta-logo"
          src="/src/images/wta-logo.png"
        />
      </a>
    </div>
    <div>
      <button
        className="button-2"
        onClick={[Function]}
        type="button"
      >
        CLE ELUM AREA1  
      </button>
    </div>
  </div>
  <div
    className="weather-weekly-chart"
  >
    <div
      className="recharts-responsive-container"
      style={
        {
          "height": 300,
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "width": 600,
        }
      }
    />
  </div>
  <div
    className="location-details-div"
    id="seattle"
  >
    <table
      className="location-details table"
    >
      <thead
        className="table-heading"
      >
        <tr
          className="secondary-heading"
        >
          <td
            className="center border-right"
            colSpan={1}
          >
            DATE
          </td>
          <td
            className="center border-right"
            colSpan={2}
          >
            DETAILS
          </td>
          <td
            className="border-right"
          >
            LOW / HIGH
          </td>
          <td
            className="center border-right"
            colSpan={2}
          >
            PRECIP
          </td>
          <td>
            CLOUD COV
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          className=" weekend "
        >
          <td
            className="border-right"
          >
            SAT
              
            17TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-day-sunny day-sunny "
              title="Clear conditions throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Clear 
          </td>
          <td
            className="border-right"
          >
            47° / 75°
          </td>
          <td
            className="align-right"
          >
            0%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            1.8%       
          </td>
        </tr>
        <tr
          className=" weekend "
        >
          <td
            className="border-right"
          >
            SUN
              
            18TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-day-sunny day-sunny "
              title="Clear conditions throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Clear 
          </td>
          <td
            className="border-right"
          >
            52° / 73°
          </td>
          <td
            className="align-right"
          >
            0%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            14.3%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            MON
              
            19TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-day-sunny day-sunny "
              title="Clear conditions throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Clear 
          </td>
          <td
            className="border-right"
          >
            52° / 70°
          </td>
          <td
            className="align-right"
          >
            0%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            18.5%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            TUE
              
            20TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-day-cloudy day-cloudy "
              title="Partly cloudy throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Partly cloudy 
          </td>
          <td
            className="border-right"
          >
            51° / 66°
          </td>
          <td
            className="align-right"
          >
            19%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            33.2%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            WED
              
            21ST
          </td>
          <td>
            <i
              className="wi weather-icon wi-day-cloudy day-cloudy "
              title="Partly cloudy throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Partly cloudy 
          </td>
          <td
            className="border-right"
          >
            49° / 65°
          </td>
          <td
            className="align-right"
          >
            5%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            31.3%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            THU
              
            22ND
          </td>
          <td>
            <i
              className="wi weather-icon wi-rain rain "
              title="Partly cloudy throughout the day with a chance of rain."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Rain
          </td>
          <td
            className="border-right"
          >
            49° / 61°
          </td>
          <td
            className="align-right"
          >
            48%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            55%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            FRI
              
            23RD
          </td>
          <td>
            <i
              className="wi weather-icon wi-rain rain "
              title="Partly cloudy throughout the day with morning rain."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Rain
          </td>
          <td
            className="border-right"
          >
            49° / 59°
          </td>
          <td
            className="align-right"
          >
            86%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            80%       
          </td>
        </tr>
        <tr
          className=" weekend "
        >
          <td
            className="border-right"
          >
            SAT
              
            24TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-rain rain "
              title="Cloudy skies throughout the day with a chance of rain."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Rain
          </td>
          <td
            className="border-right"
          >
            50° / 66°
          </td>
          <td
            className="align-right"
          >
            76%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            100%       
          </td>
        </tr>
        <tr
          className=" weekend "
        >
          <td
            className="border-right"
          >
            SUN
              
            25TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-rain rain "
              title="Cloudy skies throughout the day with a chance of rain throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Rain
          </td>
          <td
            className="border-right"
          >
            49° / 56°
          </td>
          <td
            className="align-right"
          >
            71%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            100%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            MON
              
            26TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-rain rain "
              title="Partly cloudy throughout the day with rain in the morning and afternoon."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Rain
          </td>
          <td
            className="border-right"
          >
            46° / 56°
          </td>
          <td
            className="align-right"
          >
            48%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            74.4%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            TUE
              
            27TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-cloudy cloudy sunshine-50"
              title="Partly cloudy throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Partly cloudy 
          </td>
          <td
            className="border-right"
          >
            43° / 58°
          </td>
          <td
            className="align-right"
          >
            43%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            82.5%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            WED
              
            28TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-cloudy cloudy sunshine-50"
              title="Cloudy skies throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Cloudy
          </td>
          <td
            className="border-right"
          >
            47° / 51°
          </td>
          <td
            className="align-right"
          >
            43%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            100%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            THU
              
            29TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-cloudy cloudy sunshine-50"
              title="Partly cloudy throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Partly cloudy 
          </td>
          <td
            className="border-right"
          >
            50° / 62°
          </td>
          <td
            className="align-right"
          >
            29%
          </td>
          <td
            className="align-right border-right"
          >
            0"
          </td>
          <td
            className="align-right"
          >
            89.1%       
          </td>
        </tr>
        <tr
          className=" "
        >
          <td
            className="border-right"
          >
            FRI
              
            30TH
          </td>
          <td>
            <i
              className="wi weather-icon wi-cloudy cloudy sunshine-50"
              title="Cloudy skies throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Cloudy
          </td>
          <td
            className="border-right"
          >
            51° / 60°
          </td>
          <td
            className="align-right"
          >
            24%
          </td>
          <td
            className="align-right border-right"
          >
            1"
          </td>
          <td
            className="align-right"
          >
            97.4%       
          </td>
        </tr>
        <tr
          className=" weekend "
        >
          <td
            className="border-right"
          >
            SAT
              
            1ST
          </td>
          <td>
            <i
              className="wi weather-icon wi-cloudy cloudy sunshine-10"
              title="Partly cloudy throughout the day."
            />
          </td>
          <td
            className="border-right align-left"
          >
            Partly cloudy 
          </td>
          <td
            className="border-right"
          >
            46° / 56°
          </td>
          <td
            className="align-right"
          >
            29%
          </td>
          <td
            className="align-right border-right"
          >
            1"
          </td>
          <td
            className="align-right"
          >
            55.1%       
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
